pkg/metrics/builder/builder_test.go

```diff
@@ -0,0 +1,241 @@
+package builder
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+func getDescStruct(name string, desc string, constLabels map[string]string, variableLabels []string) string {
+	lpStrings := make([]string, 0, len(constLabels))
+	for key, value := range constLabels {
+		lpStrings = append(lpStrings, fmt.Sprintf("%s=%q", key, value))
+	}
+	sort.Strings(lpStrings)
+
+	return fmt.Sprintf("Desc{fqName: %q, help: %q, constLabels: {%s}, variableLabels: %v}",
+		name, desc, strings.Join(lpStrings, ","), variableLabels)
+}
+
+func TestMetricBuilderHistogram(t *testing.T) {
+	labels := []string{"test1", "test2"}
+	labelValues := map[string]string{"test1": "test1", "test2": "test2"}
+	desc := "test histogram metrics"
+	name := "test_histogram"
+	buckets := []float64{10, 20, 30}
+
+	metricBuilder, err := NewMetricBuilder(labels,
+		labelValues,
+		desc,
+		name,
+		buckets,
+		Histogram,
+	)
+
+	if err != nil {
+		assert.Failf(t, err.Error(), "metric builder failure")
```

Comment:
assert.NoError(t, err, "expected nil error creating metric builder") something like that..

Reply:
https://godoc.org/github.com/stretchr/testify/assert#NoError
```
  actualObj, err := SomeFunction()
  if assert.NoError(t, err) {
	   assert.Equal(t, expectedObj, actualObj)
  }
```